craft_application/application.py

```diff
@@ -711,6 +711,7 @@ def _expand_environment(self, yaml_data: dict[str, Any], build_for: str) -> None
             application_name=self.app.name,  # not used in environment expansion
             cache_dir=pathlib.Path(),  # not used in environment expansion
             arch=build_for_arch,
+            parallel_build_count=util.get_parallel_build_count(self.app.name),
```

Comment:
I'm not sure this matters here?

Reply:
This ProjectInfo constructor call is what was called out in the issue for sending the parallel build count.  Where should it go, if not there?

Reply:
I think it belongs here because line 723 below is where `$CRAFT_PARALLEL_BUILD_COUNT` is evaluated in `override-` scripts.